src/DwitTech.AccountService.WebApi/Controllers/UserController.cs

```diff
@@ -1,10 +1,45 @@
-﻿using DwitTech.AccountService.Core.Dtos;
-using DwitTech.AccountService.Core.Services;
+﻿using DwitTech.AccountService.Data.Entities;
+using DwitTech.AccountService.Core.Utilities;
+using Microsoft.AspNetCore.Authorization;
 using Microsoft.AspNetCore.Mvc;
+using DwitTech.AccountService.Core.Interfaces;
+
 
 namespace DwitTech.AccountService.WebApi.Controllers
-{
+{ 
     public class UserController : BaseController
     {
+        private readonly IActivationService _activationService;
+
+        public UserController(IActivationService activationService)
+        {
+           
+            _activationService = activationService;
+        }
+
+        
+        [HttpGet("/Activation/{activationCode}")]
+        public async Task<IActionResult> ActivateUser(string activationCode)
+        {
+            try
+            {
+                string fromEmail = "support@gmail";
+                string toEmail = "info@gmail.com";
+                string templateName = "WelcomeEmail";
+                string subject = "Account Details";
+                string cc = "";
+                string bcc = "";
+
+                var activationResult = await _activationService.ActivateUser(activationCode, fromEmail , toEmail, templateName, subject, cc, bcc);
+                return Ok(activationResult);
+            }
+            catch (Exception e)
```

Comment:
I'm not sure about this, but I think you may need to handle exceptions differently here, cos It'll be very difficult to know what the error is if it returns 500 for all errors.

Reply:
> I'm not sure about this, but I think you may need to handle exceptions differently here, cos It'll be very difficult to know what the error is if it returns 500 for all errors.

Resolved